src/config.rs

```diff
@@ -1,121 +1,145 @@
 //! The `config` module provides definitions for all configuration structs as well as methods
 //! necessary for configuration via yaml and command line.
 
-/// The source configuration contains all the information necessary to build a source
-pub mod source_config {
-    use std::path::PathBuf;
-
-    /// The source configuration contains all the information necessary to build a source
-    #[derive(Deserialize, Clone)]
-    pub struct SourceConfig {
-        /// The path to the source file (relative to the configuration file, if there is one)
-        pub path: PathBuf,
-        /// The kind of the source file (image, etc) if applicable
-        pub kind: String,
-    }
-}
+use clap::{App, Arg, ArgMatches};
+use nfd::{self, Response};
+use std::collections::HashMap;
+use std::fs::File;
+use std::io::BufReader;
+use std::io::prelude::*;
+use std::path::{Path, PathBuf};
+use failure::Error;
+use failure::ResultExt;
+
+use platform::config::PlatformSpecificConfig;
 
-/// The buffer configuration contains all the information necessary to build a buffer
-pub mod buffer_config {
-    use std::path::{Path, PathBuf};
+/// Blend node operations
+#[derive(Debug, Deserialize, Clone)]
+#[allow(non_camel_case_types)]
+pub enum BlendOp {
+    /// Take the minimum RGBA value
+    min,
```

Comment:
~~I don't exactly remember, but doesn't fmt-rfc recommend PascalCased enum items?~~

Ah, I see why you're doing it this way; if you want to ditch the `#[allow(non_camel_case_types)]`, you can use `#[serde(rename_all = 'lowercase')]` (or `snake_case`) on the enum and use `PascalCase` items.

Reply:
ah, good call